src/Stratis.Features.FederatedPeg/CollateralChecker.cs

```diff
@@ -79,7 +79,7 @@ public async Task InitializeAsync()
             {
                 this.logger.LogDebug("Initializing federation member {0} with amount {1}.", federationMember.CollateralMainchainAddress, federationMember.CollateralAmount);
 
-                this.depositsByAddress.Add(federationMember.CollateralMainchainAddress, null);
+                this.depositsByAddress.Add(federationMember.CollateralMainchainAddress, 0);
```

Comment:
@zeptin whilst this change will resolve the null exception, I'm not what the initial intention was with initializing the federation member with a null deposit amount? Perhaps @noescape00 can chime in here?

Perhaps we need to throw if it is zero? Or is 0 actually a valid deposit amount?

Reply:
0 is a valid amount which means no collateral is set. 

Initially it was initialized with null because in every place where we used it we would first check if address was null